fish-rust/src/wutil/wcstoi.rs

```diff
@@ -102,11 +105,12 @@ where
     if result == 0 {
         negative = false;
     }
-    Ok(ParseResult { result, negative })
+    let consumed_all = chars.peek() == None;
+    Ok(ParseResult { result, negative, consumed_all })
 }
 
 /// Parse some iterator over Chars into some Integer type, optionally with a radix.
-fn fish_wcstoi_impl<Int, Chars>(src: Chars, mradix: Option<u32>) -> Result<Int, Error>
+fn fish_wcstoi_impl<Int, Chars>(src: Chars, mradix: Option<u32>, consume_all: bool) -> Result<Int, Error>
 where
```

Comment:
Another solution is to take a mutable iterator (like `src: &mut Chars`),
then the caller can check `src.next().is_some()`.
But that probably causes other inconveniences, I think it's fine.